### PR TITLE
Bug/include all districts of schools

### PR DESCRIPTION
**Pull request by agorina**

### CODE REVIEW

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/QueryProviderRepositoryHelper.java [hunk 1]
```diff
@@ -173,11 +173,10 @@ public QueryProviderRepositoryHelper(final OrganizationRepository organizationRe
             for (final School school : schoolsByDistrictId.get(districtId)) {
                 if (schoolIds.contains(school.getId())) {
                     schoolIdsLeftToProcess.remove(school.getId());
-                    districtIds.add(school.getId());
+                    districtIds.add(districtId);
```
Comment (agorina) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/QueryProviderRepositoryHelper.java]: doh!

File: aggregate-service/src/test/resources/OlapEntitiesSetup.sql [hunk 2]
```diff
@@ -7,15 +7,15 @@ INSERT INTO school_year VALUES (1999),(2000),(2001);
 
 -- ------------------------------------------ School/Districts --------------------------------------------------------------------------------------------------
 INSERT INTO district (id, name, natural_id, external_id, migrate_id) VALUES
-  (-7, 'District-7', 'id-7', 'externalId-7', -1),
-  (-8, 'District-8', 'id-8', 'externalId-8', -1),
-  (-9, 'District-9', 'id-9', 'externalId-9', -1);
+  (-17, 'District-7', 'id-7', 'externalId-7', -1),
+  (-18, 'District-8', 'id-8', 'externalId-8', -1),
+  (-19, 'District-9', 'id-9', 'externalId-9', -1);
```
Comment (agorina) [inline aggregate-service/src/test/resources/OlapEntitiesSetup.sql]: I was using the same numbers for schools and district IDs and this caused it to be working in the tests.

File: aggregate-service/src/test/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRowRepositoryRST.java [hunk 3]
```diff
@@ -144,6 +144,16 @@ public void itShouldChooseAllDistrictsFlagOverDistrictsOfSchools() {
...
     }
 
+    @Test
+    public void itShoulIncludeDistrictsOfSchools() {
+        assertThat(reportRepository.findByQuery(queryBuilder
+                .includeState(false)
+                .schoolIds(of(-9L, -10L))
+                .includeAllDistrictsOfSchools(true)
+                .includeAllDistricts(false)
+                .build()).size()).isEqualTo(3);
```
Comment (agorina) [inline aggregate-service/src/test/java/org/opentestsystem/rdw/olap/repository/impl/JdbcCustomAggregateReportRowRepositoryRST.java]: Added this test. It was still working until I changed the IDs to be unique between schools and districts.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/QueryProviderRepositoryHelper.java (hunk 1 above)
Comment (esotrope) [reply aggregate-service/src/main/java/org/opentestsystem/rdw/olap/sqlbuilder/QueryProviderRepositoryHelper.java]: At least you didn't add the school code x J

**Pull request merged**